R/get_precipitation_data.R

```diff
@@ -0,0 +1,149 @@
+#'
+#' To read more about CHIRPS:
+#' #' https://www.chc.ucsb.edu/data/chirps
+#' https://www.nature.com/articles/sdata201566
+#'
+#' To read more about R Google Earth Engine
+#' https://cran.r-project.org/web/packages/rgee/
+#'
+#' @param x an \code{egh_connection} object or a sf object, needed to pull the specified spatial polygon
+#' @param country a three-letter string specifying the country iso code (ISO 3166-1 alfa-3) pull data from. Must be "1", "2" or "3". Please, check https://gadm.org/ for available countries and administrative levels.
+#' @param admin_lvl a string specifying the administrative level to pull data from. Must be "1", "2" or "3". Please, check https://gadm.org/ for available countries and administrative levels.
+#' @param startDate a string containing the starting date in "yyyy-mm-dd" format for the query.
+#' @param endDate a string containing the ending date in "yyyy-mm-dd" format for the query.
+#'
+#' @section Warning:
+#' A Internet connection is needed to use this function. Longer time periods and large datasets can take several minutes to fetch data, especially if using a free account.
+#'
+#' This functions calls the rgee package. If you never used rgee, please follow the instalation instructions. You will be prompted to create an user account on Google Earth Engine or to connect to an existing one. This is mandatory for rgee to run. Please, go to https://cran.r-project.org/web/packages/rgee/ for more instructions.
+#'
+#' @examples \dontrun{
+#' library(epigraphhub)
+#' country = "AGO"
+#' admin_lvl = "2"
+#' startDate = "2021-01-01"
+#' endDate = "2022-01-11"
+#' id_col = "NAME_1"
+#' con <- egh_connection(auto_connect = TRUE, use_env = TRUE, path_env = "./.env")
+#' df <- get_precipitation_data(x = con,
+#'                              country = country,
+#'                              admin_lvl = admin_lvl,
+#'                              startDate = startDate,
+#'                              endDate = endDate)
+#' }
+#' @import rgee
+#' @import dplyr
+#' @import sf
+#' @import geojsonio
+#' @importFrom raster extent
+#' @importFrom tidyr pivot_longer
+#' @importFrom tidyselect peek_vars
+#' @export
+
+get_precipitation_data <- function(x,
+                                   id_col = NULL,
+                                   country = NULL,
+                                   admin_lvl = NULL,
+                                   startDate,
+                                   endDate){
+
+  if (missing(x)) stop("Please, provide an egh_connect() or a sf object.")
+  if (missing(startDate)) stop("Dates not provided.")
+  if (missing(endDate)) stop("Dates not provided.")
+  if (endDate < startDate) stop("Ending date cannot be equal or lower than starting date.")
+
+  # defining if user is using EGH connection or is providing a sf object
```

Comment:
@dcpcamara these codeblock described by comments in you code are good candidates to be refactored into separate functions, not only to facilitate code reuse and reduce duplication, but also to make it possible  to write specific unit tests for them.

Reply:
@fccoelho sorry but I didn't quite understand. Should I dismember these functions into multiple .R archives and then call into a new single function?